pkg/agent/nodeportlocal/nplagent_test.go

```diff
@@ -94,43 +119,239 @@ func TestMain(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	portTable = NewPortTable(mockCtrl)
 
-	c, _ := InitController(kubeClient, portTable, defaultNodeName)
+	// informerFactory is initialised and started from cmd/antrea-agent/agent.go
+	informerFactory := informers.NewSharedInformerFactory(kubeClient, resyncPeriod)
+
+	c, _ := InitController(kubeClient, informerFactory, portTable, defaultNodeName)
 	stopCh := signals.RegisterSignalHandlers()
-	go c.Worker()
-	go c.Ctrl.Run(stopCh)
-	cache.WaitForCacheSync(stopCh, c.Ctrl.HasSynced)
+
+	go c.Run(stopCh)
+	informerFactory.Start(stopCh)
+}
+
+func pollForPodAnnotation(r *require.Assertions, podName string, found bool) (string, error) {
+	var data string
+	var exists bool
+	err := wait.Poll(time.Second, 20*time.Second, func() (bool, error) {
+		updatedPod, err := kubeClient.CoreV1().Pods(defaultNS).Get(context.TODO(), podName, metav1.GetOptions{})
+		r.Nil(err, "Failed to get Pod")
+		ann := updatedPod.GetAnnotations()
+		data, exists = ann[k8s.NPLAnnotationKey]
+		if found {
+			return exists, nil
+		} else {
+			return !exists, nil
+		}
+	})
+
+	return data, err
+}
+
+// Add a service (proper annotation) and pod pair, update service with bad
```

Comment:
we capitalize K8s resources: Pod, Service, Node, etc

(several occurrences in this file)